Scripts/Source/UD_CustomDevice_RenderScript.psc

```diff
@@ -136,12 +139,64 @@ EndFunction
 ;=============================================================
 
 ;--------------------------------PUBLIC PROPERTIES----------------------------
-;-------------------------------------------------------
-;-------!!!!!!!!!!!!ALWAYS FILL!!!!!!!!!!!!!!!----------
-Armor       Property DeviceInventory                auto    ;device inventory
+
+;/  Group: Required Values
+===========================================================================================
+===========================================================================================
+===========================================================================================
+/;
+
+;/  Variable: DeviceInventory
+    *This property have to be allways filled with relevant inventory device!!*
+/;
+Armor       Property DeviceInventory                auto
+
+;/  Variable: libs
+    *Reference to zadlibs, should be allways filled!!*
+/;
 zadlibs     Property libs                           auto
-;---------------------OPTIONAL--------------------------
+
+
+;/  Group: Optional Values
+===========================================================================================
+===========================================================================================
+===========================================================================================
+/;
+
 Key         Property zad_deviceKey                  auto
+
+;/  Variable: UD_DeviceKeyword
+    Main keyword of device. In case this is not filled, it will be taken from Inventory Device
+/;
+Keyword  Property UD_DeviceKeyword                              auto hidden ;keyword of this device for better manipulation. Is taken from ID
+
+;/  Variable: DeviceRendered
+    Link to device the script is currently on. In case this is not filled, it will be taken from Inventory Device
+/;
+Armor    Property DeviceRendered                                auto hidden ;Is taken from ID
+
+
+String[] Property UD_DeviceStruggleKeywords                     auto Hidden ;keywords (as string array) used to filter struggle animations
+
+;/  Variable: UD_ActiveEffectName
+    Name of active effect. Defaultly set to "Share"
```

Comment:
Name of active effect. By default set to "Share"

Reply:
Fixed